examples/electron/test/basic-example-with-spectron.espec.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+const Application = require("spectron").Application;
+const path = require('path');
+const chai = require('chai');
+const should = chai.should();
+const chaiAsPromised = require('chai-as-promised');
+var electronPath = path.join(__dirname, '..', 'node_modules', '.bin', 'electron');
+
+// Path to your application
+var appPath = path.join(__dirname, '../src-gen/frontend/electron-main.js');
+
+import { assert } from "chai"
+const expect = chai.expect;
+const timeout = 10000;
+
+before(() => {
+    chai.config.showDiff = true;
+    chai.config.includeStack = true;
+    chai.should();
+    chai.use(chaiAsPromised);
+});
+
+const app = new Application({
+    path: electronPath,
+    args: [appPath],
+    startTimeout: timeout,
+    waitTimeout: timeout,
+    requireName: 'electronRequire'
+});
+
+describe('Test specs with spectron', function () {
+    this.timeout(60000);
+
+    beforeEach(function () {
+        require("../src-gen/backend/main"); // start the express server
```

Comment:
Should not it be done by `electron-main` and we only should wait until the window is visible?